### PR TITLE
feat: Static file system

### PR DESCRIPTION
**Pull request by andreespirela**

### CODE REVIEW

**Comment by nyannyacha** [comment]
It looks like you didn't resolve some conflicts during the rebase.

```
error[E0061]: this function takes 5 arguments but 4 arguments were supplied
   --> crates/base/tests/integration_tests.rs:169:31
    |
169 |     let (_, worker_pool_tx) = create_user_worker_pool(
    |                               ^^^^^^^^^^^^^^^^^^^^^^^
...
173 |         None,
    |         ---- an argument of type `Vec<String>` is missing
    |
note: function defined here
   --> /workspaces/edge-runtime/crates/base/src/rt_worker/worker_ctx.rs:697:14
    |
697 | pub async fn create_user_worker_pool(
    |              ^^^^^^^^^^^^^^^^^^^^^^^
help: provide the argument
    |
169 |     let (_, worker_pool_tx) = create_user_worker_pool(integration_test_helper::test_user_worker_pool_policy(), None, Some(pool_termination_token.clone()), /* Vec<String> */, None)
    |                                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error[E0063]: missing field `static_patterns` in initializer of `WorkerContextInitOpts`
   --> crates/base/tests/integration_tests.rs:178:16
    |
178 |     let opts = WorkerContextInitOpts {
    |                ^^^^^^^^^^^^^^^^^^^^^ missing `static_patterns`
```

File: examples/main-empty/index.ts [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
+import isEven from "npm:is-even";
+
+console.log('Hello A');
+globalThis.isTenEven = isEven(10);
 
 console.log(Deno.version);
 let val = sum(1, 2);
+console.log(Deno.readFile)
+console.log(Deno.cwd())
+console.log(Deno.readFileSync('examples/postgres-on-the-edge/README.md'));
```
Comment (nyannyacha) [inline examples/main-empty/index.ts:12]: In your recent changes, you seem to have decided to prefix `./mnt/data`
I think this path should be changed to reflect that change as well 😎

File: crates/cli/src/main.rs [hunk 2]
```diff
@@ -224,6 +228,14 @@ fn main() -> Result<(), anyhow::Error> {
                     sub_matches.get_one::<usize>("max-parallelism").cloned();
                 let maybe_request_wait_timeout =
                     sub_matches.get_one::<u64>("request-wait-timeout").cloned();
+                let static_patterns = sub_matches
+                    .get_many::<String>("static")
+                    .unwrap()
+                    .map(|s| s.as_str())
+                    .collect::<Vec<&str>>();
```
Comment (nyannyacha) [inline crates/cli/src/main.rs:235]: When the `--static` option is not given by the user, the process dies by panic. Is this intended?

File: crates/sb_graph/lib.rs [hunk 3]
```diff
@@ -24,6 +26,8 @@ pub mod import_map;
 
 pub const VFS_ESZIP_KEY: &str = "---SUPABASE-VFS-DATA-ESZIP---";
 pub const SOURCE_CODE_ESZIP_KEY: &str = "---SUPABASE-SOURCE-CODE-ESZIP---";
+pub const STATIC_FILES_ESZIP_KEY: &str = "---SUPABASE-STATIC-FILES-ESZIP---";
+pub const STATIC_FS_PREFIX: &str = "mnt/data";
```
Comment (laktek) [inline crates/sb_graph/lib.rs:30]: should it be `/mnt/data`?

Comment (andreespirela) [reply crates/sb_graph/lib.rs:30]: It gets normalized later on, with `normalize_path`

File: examples/main-empty/index.ts [hunk 4]
```diff
@@ -1,8 +1,15 @@
 import { serve } from "https://deno.land/std@0.131.0/http/server.ts";
 import {sum} from "./some-import.ts";
+import isEven from "npm:is-even";
+
+console.log('Hello A');
+globalThis.isTenEven = isEven(10);
 
 console.log(Deno.version);
 let val = sum(1, 2);
+console.log(Deno.readFile)
```
Comment (laktek) [inline examples/main-empty/index.ts:10]: nit: remove this

**Review by laktek — APPROVED** [review]
Please squash commit before merge.

**Pull request merged**